toytools/datasets/simple_toyzero.py

```diff
@@ -0,0 +1,150 @@
+        object will focus on.
+        Default: False
+    planes : Set[str] or None
+        A set of planes to select. If None, then all planes are selected.
+        Default: None
+    seed : int
+        Seed used to initialize random number generators.
+        Default: 0
+    transform : Callable or None,
+        Optional transformation to apply to images.
+        E.g. torchvision.transforms.RandomCrop.
+        Default: None
+    val_size : float or int
+        Fraction of the toyzero dataset that will be used as a validation
+        sample. If val_size <= 1, then it is treated as a fraction, i.e.
+        (size of val sample) = `val_size` * (size of toyzero dataset)
+        Otherwise, (size of val sample) = `val_size`.
+        Default: 0.2
+    """
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        path,
+        crop_shape       = None,
+        expansion_factor = 1,
+        is_train         = False,
+        planes           = None,
+        seed             = 0,
+        transform        = None,
+        val_size         = 0.2,
+    ):
+        super().__init__(path)
+
+        self._is_train = is_train
+        self._images   = []
+
+        self._indices   = []
+        self._len       = 0
+
+        self._crop_shape = crop_shape
+        self._transform  = transform
+        self._seed       = seed
+        self._efactor    = expansion_factor
+        self._prg        = np.random.default_rng(seed)
+        self._planes     = planes
+        self._val_size   = val_size
+
+        self._collect_images()
+        self._split_dataset()
+
+    def _collect_images(self):
+        """Locate all matching images in the dataset directory."""
+        images = collect_toyzero_images(self._path)
+        self._images = filter_images(images, None, self._planes)
+
+    def _split_dataset(self):
+        """Split images into training/validation samples."""
+        indices = np.arange(len(self._images))
+        self._prg.shuffle(indices)
+
+        val_size = self._val_size
+
+        if val_size <= 1:
+            val_size = int(len(indices) * val_size)
+        else:
+            val_size = val_size // self._efactor
+
+        train_size = len(indices) - val_size
+
+        if self._is_train:
+            self._indices = indices[:train_size]
+        else:
+            self._indices = indices[train_size:]
+
+        self._len = len(self._indices) * self._efactor
+
+    def __len__(self):
+        return self._len
+
+    def _get_rand_cropped_area(self, image_fake, image_real, bkg_value):
+        """Randomly crop images."""
+        cropped_fake_image, crop_region = try_find_region_with_signal(
+            image_fake, self._prg, bkg_value, MIN_PIXEL_DIFF,
+            self._crop_shape, CROP_RETRIES
+        )
+
+        return (cropped_fake_image, crop_image(image_real, crop_region))
+
+    def __getitem__(self, index):
+
+        if self._is_train:
+            self._prg    = np.random.default_rng()
```

Comment:
So, when `self._is_train=True`, we don't really get an entry by index, but rather a totally random entry, right?

Reply:
Yes, but I am not sure if that is a good decision. So, I am open to suggestions.

Reply:
Since we also shuffle the indices, I am wondering another random access to entry is necessary or not.

Reply:
Good point. Ok, I think I will merge this branch to speed up the process, but I will open an issue to refactor/rewrite the SimpleDataset